### PR TITLE
Prometheus endpoint

### PR DESCRIPTION
**Pull request by gaydin**
I want to try to finish the task #2751, initial commits are based on #678

~~I need help on several issues~~: 
~~1. How can l make an authorization token?
It is necessary to use a token in app.ini and make it accessible from the section /admin?
Is need to make token in modules/gererate and save in configuration,If the token doesn't exist in configuration?~~

~~2. Do I need to add "metrics" in a [reservedUsernames](https://github.com/go-gitea/gitea/blob/master/models/user.go#L685)?~~
~~3. How can l correct dependencies? 
This PR is build successfully just if update it manually github.com/golang/protobuf~~

sorry for my bad english :)

### CODE REVIEW

**Comment by gaydin** [comment]
> For the deps, does `dep ensure -update github.com/golang/protobuf` solve the problem ?

Yes it helped. thank

File: modules/metrics/collector.go
Comment (lunny) [inline modules/metrics/collector.go:5]: comment head

**Review by lunny — DISMISSED** [review]

File: modules/metrics/collector.go
Comment (techknowlogick) [inline modules/metrics/collector.go:10]: Can you add an empty line between these two imports?

**Comment by gaydin** [comment]
Bad idea to put metrics handler in a separate [file](https://github.com/go-gitea/gitea/blob/5e39a9e5a32f7909332a2c2525109d8b5174987d/routers/metrics.go)?

By default metrics are disabled. If enable metrics will be available /metrics without authorization. But you can also add a token, then access to /metrics will be available only with a token.

**Comment by techknowlogick** [comment]
I think having it in separate file is best, because it is a unique area (just like how the api is split off into different files).

**Review by lafriks — APPROVED** [review]

File: modules/setting/setting.go
Comment (techknowlogick) [inline modules/setting/setting.go:565]: Are you able to add default values to `custom/conf/app.ini.sample` and also to `docs/content/doc/advanced/config-cheat-sheet.en-us.md`

**Review by techknowlogick — CHANGES_REQUESTED** [review]

File: modules/setting/setting.go
Comment (adelowo) [inline modules/setting/setting.go]: Sounds like `Enabled` or `IsEnabled` would be a better fit for this ??

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
@lunny please review again

**Review by sapk — APPROVED** [review]

**Review by appleboy — APPROVED** [review]

**Comment by techknowlogick** [comment]
Maintainers note: Dismissing @lunny's review as requested change has been made, and there are 4 other positive reviews.

**Pull request merged**
